### PR TITLE
Integrate with the CovidNearMe.org API

### LINKED ISSUE #158: [FEATURE] Integrate with the CovidNearMe.org API
*Important:* this API is yet to be made available. This ticket will be updated with more information once that happens.

Acceptance criteria:
- [ ] A basic API client has been added
- [x] ~~Token authentication has been implemented using the endpoint provided by CovidNearMe.org~~ We were given a static API key to keep this simple.
- [ ] The Symptom Reports repository has been modified to use the CovidNearMe.org
 API rather than dummy data

Possible package to simplify API integration: https://github.com/trevorwang/retrofit.dart/

-------

# Symptom Reporting API
_Docs sent by the CovidNearMe.org Devs_

## Sending Data In

There is (now) an API token that is sent as an authorization header along with the json submission. 

## A Code Example

This API has been tested using the standard Rails test example (code below) and by using an HTTP post tool (mechanize) (belt and suspenders) specifically to verify authentication practices.

The API token has to be submitted as an Authorization header:

    url = "http://localhost:3169/api/v1/symptom_report/create"
    agent = Mechanize.new
    result = agent.post(url, symptoms, 'Authorization' => api_token)

## Verifying that Data Got There

Login to your covidnearme.org account by going to:

    https://covidnearme.org/tools/
    
and selecting the sign in link

Go to your admin dashboard by going to:

    https://covidnearme.org/admin/
    
and selecting Symptoms from the left hand sidebar

Navigate to the last symptom and look for some bit of data specific to what you entered to confirm the data you sent (for example send the word ebola in the field for what were you tested for)

## Example request
```json
{
  "user_id": "00000000-0000-0000-0000-000000000000",
  "is_fake": true,
  "location": {
    "postal_code": "10001",
    "country": "US",
  },
  "question_responses": [
    {
      "question_identifier": "shortness_of_breath",
      "response": "none",
    },
    {
      "question_identifier": "cough",
      "response": "mild",
    },
    {
      "question_identifier": "fever",
      "response": "moderate",
    },
    {
      "question_identifier": "temperature",
      "response": "37.1",
    },
    {
      "question_identifier": "nausea",
      "response": "severe",
    },
    {
      "question_identifier": "tested_for_flu_etc",
      "response": "no",
    },
    {
      "question_identifier": "tested_positive_for_flu_etc",
      "response": "yes",
    },
    {
      "question_identifier": "tested_positive_for_what",
      "response": "no",
    },
    {
      "question_identifier": "tried_to_get_tested_for_covid19",
      "response": "yes",
    },
    {
      "question_identifier": "covid19_result",
      "response": "negative",
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by ferndot**
Fixes #158 

- Integrates the symptom report flow with the CovidNearMe.org API
- Updates the symptom report model to match the required schema

### CODE REVIEW

File: lib/src/data/models/symptom_report.dart [hunk 1]
```diff
@@ -41,14 +40,13 @@ class QuestionResponse {
 
 @JsonSerializable()
 class UserLocation {
-  String zipCode;
-
   /// ISO 3166-1 alpha-2.
   String country;
+  String postalCode;
```
Comment (gspencergoog) [inline lib/src/data/models/symptom_report.dart:45]: Was the switch back from `zipCode` intentional? I know you asked me to go the other way a while back...

File: .gitignore [hunk 2]
```diff
@@ -35,3 +35,7 @@ lib/generated_plugin_registrant.dart
 
 # Exceptions to above rules.
 !/packages/flutter_tools/test/data/dart_dependencies_test/**/.packages
+
+# Custom
+.env
+!.env.example
```
Comment (gspencergoog) [inline .gitignore]: Add end of file newline.

**Review by gspencergoog — DISMISSED** [review]

File: lib/src/data/models/symptom_report.dart (hunk 1 above)
Comment (ferndot) [reply lib/src/data/models/symptom_report.dart:45]: Yeah, sorry for that. The API spec that I received required `postal_code`. So, the corresponding data model needed to be updated.

File: .gitignore (hunk 2 above)
Comment (ferndot) [reply .gitignore]: 👍

Comment (ferndot) [reply .gitignore]: Fixed!

**Review by gspencergoog — DISMISSED** [review]
LGTM!

**Comment by ferndot** [comment]
Smoketest is currently failing, but will be fixed by a larger refactor in #175.

**Comment by ferndot** [comment]
The failing iOS build is not caused by this PR. Merging!

**Pull request merged**